[Cedricaca] iP

Cedricaca opened the pull request:
# Dukey
>“Your mind is a mind” – Me (Source: ~~I made it up~~)

Dukey frees your mind of having to remember things you need to do. It's,

-text-based
-easy to learn
-~~FAST~~ SUPER FAST to use

## All you need to do is,

1. download it from [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ&ab_channel=RickAstley).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😉

And it is FREE!

## Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Level-0

Commit:
Level-1

Commit:
Level-2

Commit:
Level-3

Commit:
Level-3 improved

Commit:
Level-4

Commit:
Level-5

Commit:
Level-6

Commit:
A-TextUiTesting

Commit:
Level-7

Commit:
Level-8

Commit:
Level-8

Commit:
A-MoreOOP

Commit:
A-Packages

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Added A-JUnit tests

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Added a few related files and code for A-Jar

Commit:
Added javadocs for most methods

Commit:
Adjusted code to align with coding standard

Commit:
Added a new find feature for Level-9

Commit:
Merge tag 'A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java

Commit:
Merge tag 'Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java

Commit:
Added back features and javadocs lost during the parallel merge

yleeyilin commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,303 @@
+            currLine = currLine + 1;
+        }
+    }
+    private String find = "find";
```
Is there a need to define all these variables?

yleeyilin commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,125 @@
+            file = new File(folder, "data.txt");
+        }
+    }
+    /**
```
```suggestion

    /**
```

yleeyilin commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,125 @@
+        }
+    }
+    /**
+     * finds the input line, replaces it with String L, and rewrites the whole file
```
```suggestion
     * Finds the input line, replaces it with String L, and rewrites the whole file.
```

yleeyilin reviewed (commented):
Overall, the code is easy to read. However, readability can be improved by reducing the number of characters in a single sentence. Additionally, there can be empty lines spacing used to clearly separate the different functions. Javadocs should end with a full stop.

Commit:
Add in features to create a GUI

Commit:
Add in features required for GUI utilising fxml

Used part 4 of tutorial for GUI to ensure all features are imported successfully

Commit:
Debug the build.gradle file to ensure ShadowJar works properly with the new fxml features

Change the check() function within storage to work properly with a nonexistent file and folder

Commit:
Delete unused code and changed bye command to exit the program

Commit:
Utilise SLAP to increase code quality by breaking up code into private functions for encapsulation

SampsonYe1999 commented on src/main/java/Duke.java:
```diff
@@ -3,6 +3,9 @@
 import java.util.regex.Pattern;
 import java.util.ArrayList;
 import java.io.*;
```
Maybe you should consider listing out all the imported features from the java.io library instead of importing the whole package as a whole.

ananya21 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+    public Duke() {
+    }
+    @Override
+    public void start(Stage stage) {
```
The start(Stage stage) method is overridden but not implemented. This is the entry point for a JavaFX application where you should set up your UI elements, scene, and stage. Consider adding UI setup code here

ananya21 commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,60 @@
+    /**
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
+     */
+    private void flip() {
```
The flip method is a clever way to reverse the order of elements for different dialog box styles

ananya21 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,283 @@
+                } else {
+                    Event n = new Event(newInput, false, ldt, ldt2);
+                    taskList.add(n);
+                    String result = "OK, I have added this task :\n";
```
There is repetition in handling marked and unmarked tasks across Todos, Events, and Deadlines, consider abstracting this logic into a separate method to reduce duplication and improve maintainability.

ananya21 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,153 @@
+    @Override
+    public String toString() {
+        String s = super.toString();
+        return "[D]" + s + "(by:" + by.format(DateTimeFormatter.ofPattern("MMM d yyyy HH:ss")).replace("T", " ") +")";
```
In your toString methods for Event and Deadline, you use "MMM d yyyy HH:ss" as your format. However, it seems there is a typo: "ss" should be "mm" for minutes. The format "HH:mm" is standard for hours and minutes.

ananya21 reviewed (commented):
there is effective use of OOP principles but some code can be abstracted and simplified

sarjinius commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,23 @@
+    public String receive() {
+        return scanner.nextLine();
+    }
+    /**
+     * Displays user output as a println
+     * @param n Object that will be displayed in its toString() form
+     */
+    public void display(Object n) {
+        System.out.println(n.toString());
+    }
```
It would be better to rename the parameter "n" to "obj".

sarjinius commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,9 @@
+reset
+todo bruh
+event nig /from asd /to lalala
+deadline laugh /by tomorrow
```
Would prefer if the date/time parsing functionality is tested.

sarjinius commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,283 @@
+                if (newInput.trim().equals("")) {
+                    return "Task cannot be empty!";
+                } else {
+                    Event n = new Event(newInput, false, ldt, ldt2);
```
The name of the variable "n" would be better as "event".

sarjinius commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,153 @@
+    public void unmark() {
+        isDone = false;
+    }
+    /**
+     * Dummy function to be overridden by subclasses, specifically for writing to the storage
+     * @return String to be written to storage
+     */
+
+    public String export() {
+        return toString();
+    }
```
Why not just use toString()? It is public so it can still be accessed.

sarjinius reviewed (commented):
I like how the classes interact with each other, and the control flow was logical. Good job!

Commit:
Utilise SLAP to increase code quality by breaking up code into private functions for encapsulation

Cedricaca committed:
Merge pull request #1 from Cedricaca/A-CodeQuality

For merging

Commit:
Change Readme.md

Commit:
Change Readme.md

Commit:
Change Readme.md

Commit:
Add Javadoc and Junit

Commit:
Changed some tests to align with requirements
